src/main/java/io/strimzi/kafka/bridge/mqtt/core/MqttKafkaMapper.java

```diff
@@ -0,0 +1,134 @@
+
+                for (String templatePlaceholder : kafkaTopicTemplateParts) {
+                    if (templatePlaceholder.matches(MQTT_TOPIC_PLACEHOLDER_REGEX)) {
+                        placeholders.put(templatePlaceholder, null);
+                    }
+                }
+
+                for (String placeholderKey : mqttTopicPatternParts) {
+                    if (placeholderKey.matches(MQTT_TOPIC_PLACEHOLDER_REGEX) && placeholders.containsKey(placeholderKey)) {
+                        int index = Arrays.asList(mqttTopicPatternParts).indexOf(placeholderKey);
+                        String placeholder = mqttTopic.split("/")[index];
+                        placeholders.put(placeholderKey, placeholder);
+                    }
+                }
+
+                for (Map.Entry<String, String> entry : placeholders.entrySet()) {
+                    if (entry.getValue() != null) {
+                        mappedKafkaTopic = mappedKafkaTopic.replace(entry.getKey(), entry.getValue());
+                    } else {
+                        throw new IllegalArgumentException("The placeholder " + entry.getKey() + " was not assigned any value.");
+                    }
+                }
+
+                return mappedKafkaTopic;
+            }
+
+        }
+        return DEFAULT_KAFKA_TOPIC;
+    }
+
+    /**
+     * Helper method for Building the regex expressions for the mapping rules.
+     */
+    private void buildRegex(){
+
+        //convert the mqtt patterns to a valid regex expression.
+        // the mqtt pattern can contain placeholders like {something}, + and #.
+        // if the mqtt topic contains a +, we replace it with @singleLevelWildcardRegex
+        // if the mqtt topic contains a #, we replace it with @multiLevelWildcardRegex
+        // if the mqtt topic contains a placeholder (pattern \{\w+\}), we replace it with @placeholderRegex
+        for (MappingRule rule : this.rules) {
+            String regex = rule.getMqttTopicPattern().replaceAll(MQTT_TOPIC_PLACEHOLDER_REGEX, PLACEHOLDER_REGEX)
+                    .replace(MQTT_TOPIC_MULTI_LEVEL_WILDCARD_CHARACTER, MULTIPLE_LEVEL_WILDCARD_REGEX).replace(MQTT_TOPIC_SINGLE_LEVEL_WILDCARD_CHARACTER, SINGLE_LEVEL_WILDCARD_REGEX);
+            patterns.add(Pattern.compile(regex));
+        }
+    }
+    /**
+     * Helper method for getting the delimiter from the kafka topic template.
+     *
+     * @param kafkaTopicTemplate represents the kafka topic template.
+     * @return the delimiter used in the kafka topic template.
+     */
+    private String getKafkaTopicDelimiter(String kafkaTopicTemplate) {
+        if (kafkaTopicTemplate.contains("_")) {
+            return "_";
+        } else if (kafkaTopicTemplate.contains("-")) {
+            return "-";
+        } else if (kafkaTopicTemplate.contains(".")) {
+            return ".";
+        } else {
+            throw new IllegalArgumentException("The kafka topic template must contain a valid delimiter.");
```

Comment:
I don't see why the topic template cannot be something like `foo{myvar}` without any delimiter.

Reply:
instead of delimiters will use regex.

Reply:
There would be one more update.
For now, I am just assuming the placeholders are after / in the MQTT topic, but there can be cases the topics will look like, `user{id}/info/{myvar}`.

Reply:
I would say that this would not be a good MQTT pattern, so users could be making a mistake. Let's focus on having placeholders right after the `/` for now.

Reply:
In my r

> I don't see why the topic template cannot be something like `foo{myvar}` without any delimiter.

In my recent commit, I am handling this case, and it is working. Please take a look.
Tests like 
 `rules.add(new MappingRule("term{number}", "term/{number}"));`
 
`assertThat("Mqtt pattern term/{number} should be mapped to term{number}",
                mapper.map("term/4"), is("term4"));`
 were added.